File: Lab2/C#/Main.cs
using System.Collections.Generic;

namespace Lab2_BinaryFiles
{
    class Program
    {
        static void Main(string[] args)
        {
            string inputFileName = "input.dat";
            string outputFileName = "output.dat";
            
            List<Book> catalogue = DataReader.Reader();
            BinaryWorker.BinWriter(inputFileName, catalogue);
            List<Book> readCatalogue = BinaryWorker.BinReader(inputFileName);
            
            Printer.PrintCatalogue(readCatalogue);
            List<Book> authorsBooks = AuthorsBooks.ReceiveAuthorsBooks(readCatalogue);
            Printer.PrintNumOfBooks(authorsBooks);
    
            authorsBooks = SortBooks.SortByName(authorsBooks);
            BinaryWorker.BinWriter(outputFileName, authorsBooks);
            Printer.PrintSortedBooks(authorsBooks);
        }
    }
}

File: Lab2/Python/Main.py
from BookFile import Book
import DataReader
import BinaryWorker
import AuthorsBooks
import Printer
import SortTheBooks

inputFileName = "input.dat"
outputFileName = "output.dat"

catalogue: list[Book] = DataReader.reader()
BinaryWorker.BinWriter(inputFileName, catalogue)
readCatalogue: list[Book] = BinaryWorker.BinReader(inputFileName)

Printer.printCatalogue(readCatalogue)
authorsBooks: list[Book] = AuthorsBooks.receiveAuthorsBooks(readCatalogue)
Printer.printNumOfBooks(authorsBooks)

sortedBooks = SortTheBooks.SortBooks(authorsBooks)
BinaryWorker.BinWriter(outputFileName, sortedBooks)
Printer.printSortedBooks(authorsBooks)
